frame/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+//! # FRAME
+//!
+//! > Substrate's State Transition Function (Runtime) Framework.
+//!
+//!   ______   ______    ________   ___ __ __   ______
+//!  /_____/\ /_____/\  /_______/\ /__//_//_/\ /_____/\
+//!  \::::_\/_\:::_ \ \ \::: _  \ \\::\| \| \ \\::::_\/_
+//!   \:\/___/\\:(_) ) )_\::(_)  \ \\:.      \ \\:\/___/\
+//!    \:::._\/ \: __ `\ \\:: __  \ \\:.\-/\  \ \\::___\/_
+//!     \:\ \    \ \ `\ \ \\:.\ \  \ \\. \  \  \ \\:\____/\
+//!      \_\/     \_\/ \_\/ \__\/\__\/ \__\/ \__\/ \_____\/
+//!
+//!
+//! # Introduction
+//!
+//! Substrate is at a very high level composed of two parts:
+//!
+//! 1. A *runtime* who's representing the state transition function of a blockchain, and is encoded
+//! as a wasm blob.
+//! 2. A client' who's primary purpose is to execute the given runtime.
+//!
+//! FRAME is the primary framework to build a runtime.
+//!
+//! ## Pallets
+//!
+//! ```
+//! use frame::pallet;
+//!
+//! #[pallet::pallet]
+//! mod pallet {
+//!
+//! 	#[pallet::config]
+//! 	pub trait Config: frame_system::Config {}
+//!
+//! 	#[pallet::pallet]
+//! 	pub struct Pallet<T>(_);
+//! }
+//! # fn main() {}
+//! ```
+//!
+//! ## Runtime
+//!
+//! <TODO>
+//!
+//! # This Crate
+//!
+//! This crate is an amalgamation of multiple other crates that are often used together to compose a
```

Comment:
```
//! This crate amalgamates multiple other crates that are frequently used together to compose a
//! pallet. However, its use is not mandatory and it might not meet all needs.
//!
//! In essence, this crate re-exports types and traits from various sources. All of these
//! sources are listed and re-exported again in [deps].
```